scripts/script.js

```diff
@@ -60,48 +60,109 @@ export default function () {
-  bytesPerMsBoost.add(contentLength / response.timings.duration)
+  if (response.headers["Content-Length"] !== undefined) {
+    let contentLength = parseInt(response.headers['Content-Length'])
+    if (!isNan(contentLength)) {
+      dataReceivedBoost.add(contentLength, { url: response.url })
+      bytesPerMsBoost.add(contentLength / response.timings.duration)
+    }
+  }
 
   return response
 }
 
+/**
+ * Fetches a piece CID from the RAW_FETCH_URL
+ * @param {string} piece The piece CID string to fetch
+ * @returns A K6 HTTP response from the RAW_FETCH_URL (https://k6.io/docs/javascript-api/k6-http/response/)
+ */
 function fetchFromRawUrl(piece) {
   if (__ENV.RAW_FETCH_URL) {
-    let response = http.get(`${__ENV.RAW_FETCH_URL}${piece}`, {
+    let response = get(`${__ENV.RAW_FETCH_URL}${piece}`, {
       tags: {
         name: 'RawFetchURL',
-      },
-      timeout: `${__ENV.SIMULTANEOUS_DOWNLOADS}h`,
+      }
     })
     timeRaw.add(response.timings.duration)
     ttfbRaw.add(response.timings.waiting)
     rawSuccess.add(response.status >= 200 && response.status < 300)
 
-    let contentLength = parseInt(response.headers['Content-Length'])
-    dataReceivedRaw.add(contentLength, { url: response.url })
-    bytesPerMsRaw.add(contentLength / response.timings.duration)
+    if (response.headers["Content-Length"] !== undefined) {
+      let contentLength = parseInt(response.headers['Content-Length'])
+      if (!isNan(contentLength)) {
+        dataReceivedRaw.add(contentLength, { url: response.url })
+        bytesPerMsRaw.add(contentLength / response.timings.duration)
+      }
+    }
 
     return response
   }
 }
 
-const TEST_NAME = "script";
+/**
+ * Wraps K6 http.get() to provide default and optional request parameters
+ * @param {string} url The url to fetch from
+ * @param {Params} [params] Default K6 request parameters to use. https://k6.io/docs/javascript-api/k6-http/params/
+ * @returns A K6 HTTP response (https://k6.io/docs/javascript-api/k6-http/response/)
+ */
+function get(url, params) {
+  // Ensure params is not undefined
+  if (params === undefined) params = {}
+
+  // Default timeout
+  if(__ENV.SIMULTANEOUS_DOWNLOADS != undefined) {
+    params.timeout = `${__ENV.SIMULTANEOUS_DOWNLOADS}h`
+  }
+
+  // Get random range offset and create Range header
+  if (__ENV.RANGE_SIZE != undefined) {
+    if (params.headers === undefined) params.headers = {} // Ensure headers are not undefined
+    params.headers["Range"] = getRangeHeaderValue(parseInt(__ENV.RANGE_SIZE))
+  }
 
+  return http.get(`${url}`, params)
+}
+
+/**
+ * Gets a random HTTP Range header byte value of size rangSize within maxContentSize
+ * @param {number} rangeSize The size of the content to fetch in bytes
+ * @param {number} [maxContentSize] The max size of the content in bytes. Defaults to 32GB.
+ * @returns An HTTP Range header byte value
+ */
+function getRangeHeaderValue(rangeSize, maxContentSize=32768000) {
```

Comment:
what's `32768000`? shouldn't it be `34359738368`? or am I missing something?

Reply:
Nope, I just did some bad math!